streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -129,50 +129,52 @@ public void register(StateStore store, boolean loggingEnabled, StateRestoreCallb
         }
 
-        // block until the partition is ready for this state changelog topic or time has elapsed
-        int partition = getPartition(topic);
-        boolean partitionNotFound = true;
-        long startTime = System.currentTimeMillis();
-        long waitTime = 5000L;      // hard-code the value since we should not block after KIP-4
-
-        do {
-            try {
-                Thread.sleep(50L);
-            } catch (InterruptedException e) {
-                // ignore
-            }
+        if (topic != null) {
+            // block until the partition is ready for this state changelog topic or time has elapsed
+            int partition = getPartition(topic);
+            boolean partitionNotFound = true;
+            long startTime = System.currentTimeMillis();
+            long waitTime = 5000L;      // hard-code the value since we should not block after KIP-4
+
+            do {
+                try {
+                    Thread.sleep(50L);
+                } catch (InterruptedException e) {
+                    // ignore
+                }
 
-            List<PartitionInfo> partitionInfos = restoreConsumer.partitionsFor(topic);
-            if (partitionInfos == null) {
-                throw new StreamsException("Could not find partition info for topic: " + topic);
-            }
-            for (PartitionInfo partitionInfo : partitionInfos) {
-                if (partitionInfo.partition() == partition) {
-                    partitionNotFound = false;
-                    break;
+                List<PartitionInfo> partitionInfos = restoreConsumer.partitionsFor(topic);
+                if (partitionInfos == null) {
+                    throw new StreamsException("Could not find partition info for topic: " + topic);
                 }
-            }
-        } while (partitionNotFound && System.currentTimeMillis() < startTime + waitTime);
+                for (PartitionInfo partitionInfo : partitionInfos) {
+                    if (partitionInfo.partition() == partition) {
+                        partitionNotFound = false;
+                        break;
+                    }
+                }
+            } while (partitionNotFound && System.currentTimeMillis() < startTime + waitTime);
 
-        if (partitionNotFound)
-            throw new StreamsException("Store " + store.name() + "'s change log (" + topic + ") does not contain partition " + partition);
+            if (partitionNotFound)
+                throw new StreamsException("Store " + store.name() + "'s change log (" + topic + ") does not contain partition " + partition);
+        }
 
         this.stores.put(store.name(), store);
 
-        if (isStandby) {
-            if (store.persistent())
-                restoreCallbacks.put(topic, stateRestoreCallback);
-        } else {
-            restoreActiveState(topic, stateRestoreCallback);
+        if (topic != null) {
+            if (isStandby) {
+                if (store.persistent())
```

Comment:
my preference is to always use `{..}` for `if` . Without them it reminds me of the goto fail bug!